addons/mail/static/src/discuss/autogrow_input.js

```diff
@@ -0,0 +1,98 @@
+/** @odoo-module */
+
+import { useFocus, useHover } from "../utils";
+import {
+    Component,
+    useRef,
+    useState,
+    useExternalListener,
+    onWillUpdateProps,
+} from "@odoo/owl";
+
+export class AutogrowInput extends Component {
+    setup() {
+        this.state = useState({
+            value: this.props.value,
+        });
+        this.inputFocus = useFocus("input");
+        this.inputRef = useRef("input");
+        this.rootHover = useHover("root");
+        onWillUpdateProps((nextProps) => {
+            if (this.props.value !== nextProps.value) {
+                this.state.value = nextProps.value;
+                this.inputRef.el.value = nextProps.value;
+            }
+        });
```

Comment:
`props.value` should be used as the default value for `state.value`, but you should not attempting syncing `props.value` and `state.value`.

Reason is that any change of props will re-sync, and it'll make editing the `input` annoying.

Don't worry about change of thread: it's a non-issue because all `<Thread>` define a proper `t-key`, so it will unmount/mount `AutogrowInput`